functions/src/export-geojson.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2025 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as functions from 'firebase-functions';
+import {Map} from 'immutable';
+import JSONStream from 'jsonstream-ts';
+import {canExport, canImport} from './common/auth';
+import {getDatastore} from './common/context';
+import * as HttpStatus from 'http-status-codes';
+import {DecodedIdToken} from 'firebase-admin/auth';
+import {toMessage} from '@ground/lib';
+import {GroundProtos} from '@ground/proto';
+import {toGeoJsonGeometry} from '@ground/lib';
+
+import Pb = GroundProtos.ground.v1beta1;
+
+/**
+ * Iterates over all LOIsin a job returning a valid GeoJSON file.
+ */
+export async function exportGeojsonHandler(
+  req: functions.Request,
+  res: functions.Response<any>,
+  user: DecodedIdToken
+) {
+  const db = getDatastore();
+  const {uid: userId} = user;
+  const surveyId = req.query.survey as string;
+  const jobId = req.query.job as string;
+  const surveyDoc = await db.fetchSurvey(surveyId);
+  if (!surveyDoc.exists) {
+    res.status(HttpStatus.NOT_FOUND).send('Survey not found');
+    return;
+  }
+  if (!canExport(user, surveyDoc)) {
+    res.status(HttpStatus.FORBIDDEN).send('Permission denied');
+    return;
+  }
+  const ownerId = canImport(user, surveyDoc) ? undefined : userId;
+
+  console.log(
+    `Exporting survey '${surveyId}', job '${jobId}', owner '${
+      ownerId || 'survey organizer'
+    }'`
+  );
+
+  const jobDoc = await db.fetchJob(surveyId, jobId);
+  if (!jobDoc.exists || !jobDoc.data()) {
+    res.status(HttpStatus.NOT_FOUND).send('Job not found');
+    return;
+  }
+  const job = toMessage(jobDoc.data()!, Pb.Job);
+  if (job instanceof Error) {
+    res
+      .status(HttpStatus.INTERNAL_SERVER_ERROR)
+      .send('Unsupported or corrupt job');
+    return;
+  }
+  const {name: jobName} = job;
+
+  res.type('application/json');
+  res.setHeader(
+    'Content-Disposition',
+    'attachment; filename=' + getFileName(jobName)
+  );
+  const jsonStream = JSONStream.stringify(
```

Comment:
Out of curiosity, why does this need to be defined as a string, with all the newlines? Make it a bit difficult to read. Maybe you could add a comment explaining this?

Reply:
/**
   * Creates a JSON stream representing a GeoJSON FeatureCollection.
   * The arguments follow a pattern of open, separator, close, and
   * indent, which helps to organize the JSON structure for clarity.
   */